dandi/dandiapi.py

```diff
@@ -325,24 +349,82 @@ class Version(APIBase):
     modified: datetime
 
 
-class RemoteDandiset(APIBase):
+class RemoteDandiset:
     """
     Representation of a Dandiset (as of a certain version) retrieved from the
     API
     """
 
-    client: "DandiAPIClient"
-    identifier: str
-    created: datetime
-    modified: datetime
-    #: The version in question of the Dandiset
-    version: Version
-    most_recent_published_version: Optional[Version]
-    draft_version: Version
+    def __init__(
+        self,
+        client: DandiAPIClient,
+        identifier: str,
+        version: Union[str, Version, None],
+        data: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        self.client = client
+        self.identifier = identifier
+        self._version_id: Optional[str]
+        self._version: Optional[Version]
+        if version is None:
+            self._version_id = None
```

Comment:
needs a test since I am not sure what would happen whenever I try to instantiate without providing a version - would it be a draft one? last released? or some error? if I start performing some actions.
only looking below saw it being lazily populated in `@version_id` -- worth a comment here on that.

Reply:
First of all, end users shouldn't be calling the constructors for any of the API resource classes.  If no version is specified, it follows the same rules as documented under `DandiAPIClient.get_dandiset()`.

Reply:
Covered by tests.

Reply:
> end users shouldn't be calling the constructors for any of the API resource classes

filed a dedicated https://github.com/dandi/dandi-cli/issues/731 since such separation was not clear to me at all